src/parser.cc

```diff
@@ -1252,10 +1273,15 @@ bool readXml(tinyxml2::XMLElement *_xml, ElementPtr _sdf,
       if (descCounter == _sdf->GetElementDescriptionCount()
             && std::strchr(elemXml->Value(), ':') == nullptr)
       {
-        sdfdbg << "XML Element[" << elemXml->Value()
+        std::stringstream ss;
+        ss << "XML Element[" << elemXml->Value()
                << "], child of element[" << _xml->Value()
                << "], not defined in SDF. Copying[" << elemXml->Value() << "] "
                << "as children of [" << _xml->Value() << "].\n";
+
+        addRecoverableWarning(
```

Comment:
Note, for now this changes the default behavior for this specific check back to warn. However, this function takes in a policy enum value so it could be overridden for the default case with something like:

```
sdf::WarningsPolicy policy = _config.WarningsPolicy();
if (policy != sdf::WarningsPolicy::PEDANTIC)
  policy = sdf::WarningsPolicy::IGNORED;
```

It might also be desired to separate WarningsPolicy and add something specific like `_config.IgnoreUnrecognizedElements()`

Reply:
Since we've gone back and forth on this behavior in the past, I would vote for having `_config.IgnoreUnrecognizedElements()` or something similar.

Reply:
Yah, the more I think about it that makes sense. I switched this back to warn and I'll open a separate PR for easy review of the new additions.

Reply:
On second thought, after working on it I decided to update this PR. I think there is enough overlap between the two policies that they should have a common enum, which necessitated changes here.